mne/viz/evoked.py

```diff
@@ -1529,162 +1689,204 @@ def plot_compare_evokeds(evokeds, picks=list(), gfp=False, colors=None,
         return figs
-    else:
-        ch_type = ch_types[0]
-        ymin, ymax = ylim.get(ch_type, [None, None])
 
-    # deal with dict/list of lists and the CI
-    if ci is not None and not (isinstance(ci, np.float) or callable(ci)):
-        raise TypeError('ci must be float or callable, got ' + str(type(ci)))
-
-    scaling = _handle_default("scalings")[ch_type]
-    unit = _handle_default("units")[ch_type]
+    # From now on there is only 1 channel type
+    assert len(ch_types) == 1
+    ch_type = ch_types[0]
 
     all_positive = gfp  # True if not gfp, False if gfp
-    if ch_type == 'grad' and len(picks) > 1:  # deal with grad pairs
-        from ..channels.layout import _merge_grad_data
+    pos_picks = picks  # keep locations to pick for plotting
+    if ch_type == "grad" and len(picks) > 1:
+        logger.info('Combining all planar gradiometers with RMSE.')
+        pos_picks, _ = _grad_pair_pick_and_name(one_evoked.info, picks)
+        pos_picks = pos_picks[::2]
         all_positive = True
-        if gfp is not True:
-            picks, ch_names = _grad_pair_pick_and_name(example.info, picks)
+        for cond, this_evokeds in evokeds.items():
+            evokeds[cond] = [_combine_grad(e, picks) for e in this_evokeds]
+        ch_names = evokeds[cond][0].ch_names
+        picks = range(len(ch_names))
+
+    del info
+
+    ymin, ymax = ylim.get(ch_type, [None, None])
+
+    scaling = _handle_default("scalings")[ch_type]
+    unit = _handle_default("units")[ch_type]
 
     if (ymin is None) and all_positive:
         ymin = 0.  # 'grad' and GFP are plotted as all-positive
 
     # if we have a dict/list of lists, we compute the grand average and the CI
-    if ci is None:
-        ci = False
-    if not all([isinstance(evoked_, Evoked) for evoked_ in evokeds.values()]):
-        if ci is not False:
-            if callable(ci):
-                _ci_fun = ci
-            else:
-                from ..stats import _ci
-                _ci_fun = partial(_ci, ci=ci, method="bootstrap")
-            # calculate the CI
-            ci_array = dict()
-            for condition in conditions:
-                # this will fail if evokeds do not have the same structure
-                # (e.g. channel count)
-                data = np.asarray([evoked_.data[picks, :].mean(0)
-                                   for evoked_ in evokeds[condition]])
-                ci_array[condition] = _ci_fun(data) * scaling
-
-        # get the grand mean
-        evokeds = dict((cond, combine_evoked(evokeds[cond], weights='equal'))
-                       for cond in conditions)
-    else:
-        ci = False
+    _ci_fun = None
+    if ci is not False:
+        if callable(ci):
+            _ci_fun = ci
+        else:
+            from ..stats import _ci
+            _ci_fun = partial(_ci, ci=ci, method="bootstrap")
+
+    # calculate the CI
+    ci_dict = dict()
+    data_dict = dict()
+    for cond, this_evokeds in evokeds.items():
```

Comment:
order here is not deterministic, probably better to do `for cond in sorted(evokeds.keys()): this_evokeds = evokeds[key] ...`